### PR TITLE
Increase default trie cache size to 1GiB

### PR DESCRIPTION
**Pull request by bkchr**
The default trie cache size before was set to `64MiB`, which is quite low to achieve real speed ups. `1GiB` should be a reasonable number as the requirements for validators/collators/full nodes are much higher when it comes to minimum memory requirements. Also the cache will not use `1GiB` from the start and fills over time. The setting can be changed by setting `--trie-cache-size BYTE_SIZE`.

### CODE REVIEW

**Review by iulianbarbu — APPROVED** [review]

File: prdoc/pr_6546.prdoc [hunk 1]
```diff
@@ -0,0 +1,12 @@
+title: Increase default trie cache size to 1GiB
+doc:
+- audience: Node Operator
+  description: "The default trie cache size before was set to `64MiB`, which is quite\
+    \ low to achieve real speed ups. `1GiB` should be a reasonable number as the requirements\
+    \ for validators/collators/full nodes are much higher when it comes to minimum\
+    \ memory requirements. Also the cache will not use `1GiB` from the start and fills\
+    \ over time. The setting can be changed by setting `--trie-cache-size BYTE_SIZE`.\r\
```
Comment (michalkucharczyk) [inline prdoc/pr_6546.prdoc:8]: nit: removal of state_cache_size could be mentioned.

**Review by michalkucharczyk — APPROVED** [review]

File: prdoc/pr_6546.prdoc [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    \n"
+crates:
+- name: sc-cli
+  bump: patch
```
Comment (sandreim) [inline prdoc/pr_6546.prdoc:12]: Doesn't removal of `state_cache_size` require a major here ?

**Review by sandreim — APPROVED** [review]

File: prdoc/pr_6546.prdoc (hunk 2 above)
Comment (bkchr) [reply prdoc/pr_6546.prdoc:12]: The field wasn't public anymore.

**Review by lexnv — APPROVED** [review]

File: prdoc/pr_6546.prdoc (hunk 1 above)
Comment (bkchr) [inline prdoc/pr_6546.prdoc:8]:
```suggestion
    \ over time. The setting can be changed by setting `--trie-cache-size BYTE_SIZE`.\
    The CLI option `--state-cache-size` is also removed, which was not having any effect anymore.\r\
```

**Pull request merged**